.github/workflows/build.yml

```diff
@@ -0,0 +1,84 @@
+name: Build, Test and Deploy
+
+on:
+  push:
+    branches:
+      - master
+      - beta
+  pull_request:
+
+
+jobs:
+  lint:
+    name: Linting
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
+    steps:
+      - name: Cancel previous running workflows
+        uses: fkirc/skip-duplicate-actions@master
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v1
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - name: install dependencies
+        run: yarn install --frozen-lockfile --non-interactive --production=false
+      - name: Lint commit message
+        uses: wagoid/commitlint-github-action@v2
+      - name: lint Javascript
+        run: yarn lint
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    needs: [lint]
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v1
+      - name: Cache node_modules
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - name: Send coverage
+        uses: paambaati/codeclimate-action@v2.7.4
+        env:
+          CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
+        with:
+          coverageCommand: yarn test:coverage
+
+  deploy:
+    name: Release package
+    runs-on: ubuntu-latest
+    needs: [test]
+    if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/beta')
```

Comment:
Will there be 2 releases without the "[skip ci]" condition?

Reply:
Nop because test is executed only if there is no `[skip ci]` so as deploy `needs` test it won't start.

Reply:
I understand why you said that the if on test is actually not necessary it's just in case one day someone as the idea to launch them in parallel but I can remove it if you want